### PR TITLE
Blocking the use of semver2 versions in the pack command.

### PR DESCRIPTION
**Pull request by toddm**

### CODE REVIEW

File: src/NuGet.Core/NuGet.Packaging/PackageCreation/Resources/NuGetResources.resx [hunk 1]
```diff
@@ -186,4 +186,7 @@
   <data name="Manifest_ReferencesIsEmpty" xml:space="preserve">
     <value>The element package\metadata\references\group must contain at least one &lt;reference&gt; child element.</value>
   </data>
+  <data name="SemVer2VersionsNotSupported" xml:space="preserve">
+    <value>Versions using semver2 are not supported: {0}.</value>
```
Comment (emgarten) [inline src/NuGet.Core/NuGet.Packaging/PackageCreation/Resources/NuGetResources.resx]: SemVer 2.0.0 :smile:

File: src/NuGet.Core/NuGet.Versioning/NuGetVersion.cs [hunk 2]
```diff
@@ -184,5 +184,13 @@ public int Revision
         {
             get { return _version.Revision; }
         }
+
+        /// <summary>
+        /// Returns true is version is a semver2 version
```
Comment (emgarten) [inline src/NuGet.Core/NuGet.Versioning/NuGetVersion.cs]: typo after true, here also it should probably use the official SemVer 2.0.0 name

**Comment by emgarten** [comment]
:shipit:

File: src/NuGet.Core/NuGet.Packaging/PackageCreation/Authoring/PackageBuilder.cs [hunk 3]
```diff
@@ -267,6 +267,11 @@ public void Save(Stream stream)
                 throw new InvalidOperationException(NuGetResources.SemVerSpecialVersionTooLong);
             }
 
+            if (Version != null && Version.IsSemVer2)
+            {
+                throw new InvalidOperationException(String.Format(CultureInfo.CurrentCulture, NuGetResources.SemVer2VersionsNotSupported, Version.ToString()));
```
Comment (joelverhagen) [inline src/NuGet.Core/NuGet.Packaging/PackageCreation/Authoring/PackageBuilder.cs]: nit: the `Version.ToString()` can be `Version` right? The `.ToString()` should be implicit in `String.Format`.

File: src/NuGet.Core/NuGet.Packaging/PackageCreation/Resources/NuGetResources.resx [hunk 4]
```diff
@@ -186,4 +186,7 @@
   <data name="Manifest_ReferencesIsEmpty" xml:space="preserve">
     <value>The element package\metadata\references\group must contain at least one &lt;reference&gt; child element.</value>
   </data>
+  <data name="SemVer2VersionsNotSupported" xml:space="preserve">
+    <value>Versions using SemVer 2.0.0.0 are not supported: {0}.</value>
```
Comment (joelverhagen) [inline src/NuGet.Core/NuGet.Packaging/PackageCreation/Resources/NuGetResources.resx]: nit: the extra `.0` at the end of the SemVer version is not standard. See heading:
http://semver.org/

Comment (toddm) [inline src/NuGet.Core/NuGet.Packaging/PackageCreation/Resources/NuGetResources.resx]: Oops.  That was a typo.  Fixing it now.

File: src/NuGet.Core/NuGet.Versioning/NuGetVersion.cs [hunk 5]
```diff
@@ -184,5 +184,13 @@ public int Revision
         {
             get { return _version.Revision; }
         }
+
+        /// <summary>
+        /// Returns true if version is a SemVer 2.0.0 version
+        /// </summary>
+        public bool IsSemVer2
+        {
+            get { return this.ReleaseLabels.Count() > 1 || this.HasMetadata; }
```
Comment (joelverhagen) [inline src/NuGet.Core/NuGet.Versioning/NuGetVersion.cs:193]: I'm not a SemVer buff -- so having a single release label is acceptable for SemVer 1.0.0, but more than one is not okay?

**Comment by joelverhagen** [comment]
:shipit:

File: src/NuGet.Core/NuGet.Versioning/NuGetVersion.cs (hunk 5 above)
Comment (toddm) [inline src/NuGet.Core/NuGet.Versioning/NuGetVersion.cs:193]: Yes, that is the way it works.  A second ReleaseLabel means it's using version 2.0.0.

**Pull request closed without merging**